lib/ocpp/v201/charge_point.cpp

```diff
@@ -2047,19 +2047,58 @@ void ChargePoint::handle_get_base_report_req(Call<GetBaseReportRequest> call) {
     }
 }
 
-void ChargePoint::handle_get_report_req(Call<GetReportRequest> call) {
+void ChargePoint::handle_get_report_req(const EnhancedMessage<v201::MessageType>& message) {
+    Call<GetReportRequest> call = message.call_message;
     const auto msg = call.msg;
-
+    std::vector<ReportData> report_data;
     GetReportResponse response;
 
-    // TODO(piet): Propably split this up into several NotifyReport.req depending on ItemsPerMessage /
-    // BytesPerMessage
-    const auto report_data = this->device_model->get_report_data(ReportBaseEnum::FullInventory, msg.componentVariable,
-                                                                 msg.componentCriteria);
-    if (report_data.empty()) {
-        response.status = GenericDeviceModelStatusEnum::EmptyResultSet;
-    } else {
-        response.status = GenericDeviceModelStatusEnum::Accepted;
+    const auto max_items_per_message =
+        this->device_model->get_value<int>(ControllerComponentVariables::ItemsPerMessageGetReport);
+    const auto max_bytes_per_message =
+        this->device_model->get_value<int>(ControllerComponentVariables::BytesPerMessageGetReport);
+
+    // B08.FR.17
+    if (msg.componentVariable.has_value() and msg.componentVariable->size() > max_items_per_message) {
+        // send a CALLERROR
+        const auto call_error = CallError(call.uniqueId, "OccurenceConstraintViolation", "", json({}));
+        this->send(call_error);
+        return;
+    }
+
+    // B08.FR.18
+    if (message.message_size > max_bytes_per_message) {
+        // send a CALLERROR
+        const auto call_error = CallError(call.uniqueId, "FormatViolation", "", json({}));
+        this->send(call_error);
+        return;
+    }
+
+    // if a criteria is not supported then send a not supported response.
+    auto sup_criteria =
+        this->device_model->get_optional_value<std::string>(ControllerComponentVariables::SupportedCriteria);
```

Comment:
What happens if this optional value is not implemented? What do you want to happen?